cmd/otel-agent/subcommands/run/command.go

```diff
@@ -100,7 +100,10 @@ func runOTelAgentCommand(ctx context.Context, params *subcommands.GlobalParams,
 	if err == agentConfig.ErrNoDDExporter {
 		return fxutil.Run(
 			fx.Supply(uris),
-			fx.Supply(optional.NewNoneOption[coreconfig.Component]()),
+			fx.Provide(func() (coreconfig.Component, error) {
+				pkgconfigenv.DetectFeatures(acfg)
```

Comment:
Which component requires this ?

Reply:
removed for now, currently it is not needed when there's no DD exporter config